### PR TITLE
[WIP] Add Checks to tc-gh

### PR DESCRIPTION
**Pull request by owlishDeveloper**

### CODE REVIEW

**Comment by owlishDeveloper** [comment]
Actually, looking at this - I will be upgrading `github` dependency in a separate PR which I will open tomorrow.

File: src/api.js [hunk 1]
```diff
@@ -187,7 +186,7 @@ builder.declare({
   stability:  'experimental',
   description: [
     'Capture a GitHub event and publish it via pulse, if it\'s a push,',
-    'release or pull request.',
+    'release or pull request. We also publish ping and checks events',
```
Comment (djmitche) [inline src/api.js]: We do not seem to actually publish ping events..

File: schemas/v1/github-checkrun-message.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+$schema:  http://json-schema.org/draft-06/schema#
+title:        "GitHub Check Run Message"
+description: |
+  Message reporting that a GitHub check run has occurred
+type:         object
+
+properties:
```
Comment (djmitche) [inline schemas/v1/github-checkrun-message.yml]: Shouldn't this have something about the check that ran, too?

File: src/api.js [hunk 3]
```diff
@@ -286,16 +295,17 @@ builder.declare({
...
   // Not all webhook payloads include an e-mail for the user who triggered an event
-  let headUser = msg.details['event.head.user.id'].toString();
+  let headUser = msg.body.sender.id.toString();
   let userDetails = (await instGithub.users.getById({id: headUser})).data;
-  msg.details['event.head.user.email'] = this.ajv.validate({type: 'string', format: 'email'}, userDetails.email) ?
-    userDetails.email :
-    msg.details['event.head.user.login'].replace(/\[bot\]$/, '') + '@users.noreply.github.com';
+  msg.details['event.head.user.email'] = this.ajv.validate({type: 'string', format: 'email'}, userDetails.email)
+    ? userDetails.email
+    : msg.body.sender.login.replace(/\[bot\]$/, '') + '@users.noreply.github.com';
```
Comment (djmitche) [inline src/api.js:302]: This bit looks unrelated to checks?

File: src/api.js [hunk 4]
```diff
@@ -505,7 +515,7 @@ builder.declare({
     } catch (e) {
       // 403 from Github indicates this integration doesn't have permission to post this status,
       // so return that on to the user
-      if (e.code == 403) {
+      if (e.code === 403) {
```
Comment (djmitche) [inline src/api.js]: it'd be good to land all of these fixes (===, extra semicolons, etc.) in a single PR.  I suspect your editor is auto-fixing these?  They're distracting and have the potential to cause merge conflicts when mixed into other, more substantive PRs.

**Review by djmitche — COMMENTED** [review]
It looks like the overall idea here is just to get check run webhooks "wired up" to sending pulse messages -- but not actually do anything with those pulse messages?  That seems reasonable, but what is the bigger picture that this fits into?

File: schemas/v1/github-checkrun-message.yml (hunk 2 above)
Comment (owlishDeveloper) [reply schemas/v1/github-checkrun-message.yml]: yeah, this part is not finished yet

File: src/api.js (hunk 4 above)
Comment (owlishDeveloper) [reply src/api.js]: No, it's me who auto-fixes these. Yeah, I can totally land these in a separate PR (I'll throw couple of other similar things in that PR then, actually)

Comment (djmitche) [reply src/api.js]: We could certainly enable a few more eslint checks across the board (for all of our JS repos) to prevent re-adding errors like these.  But that's a whole different project :)

File: src/api.js (hunk 3 above)
Comment (owlishDeveloper) [reply src/api.js:302]: Yeah, you're right - I'll gather these to a separate PR.

File: src/api.js (hunk 4 above)
Comment (owlishDeveloper) [reply src/api.js]: Adding couple of rules to `eslint-config-taskcluster` seems like an easy thing to do? I can do that

File: src/api.js (hunk 3 above)
Comment (owlishDeveloper) [reply src/api.js:302]: so, since most of the syntax changes in this PR are left for a "good first bug", and there's no eslint rule that enforces ternary operator formatting, I'll be leaving this particular change in this PR

**Comment by owlishDeveloper** [comment]
phew, cleaned up this PR 😅 Now all the irrelevant things are gone 👍

**Comment by owlishDeveloper** [comment]
The bigger picture here is that after we got the `check_suite` message out of the pulse, we will be creating check runs for each of the tasks. Those check runs will be added to the suite by github, and then it should Just Work... or something along these lines. I would try to avoid any concurrency though. I'm going to see if I can re-animate tc-cat and play with it a bit tomorrow to examine these webhooks a bit closer

**Comment by owlishDeveloper** [comment]
I'm going to close this PR for a while, so that it doesn't bother anybody.
